weatherbot/main.py

```diff
@@ -33,29 +33,26 @@ def parse_cli(args):
 
 
 def main():
-    data = get_updates()
-    update_id = get_update_id(data)
+    running = True
+
+    while running:
 
-    while True:
         data = get_updates()
-        update_id = get_update_id(data)
         chat_id = get_chat_id(data)
-        
-        try:
-            message_text = int(get_message(data))
+        message_text = get_message(data)
 
-        except ValueError:
+        try:
             weather = get_weather(message_text, OWN_TOKEN)
+
+        except KeyError:
```

Comment:
It means that `get_weather` can throw a `KeyError`. This kind of behavior is very implicit and opaque. You need to specify it in `get_weather` comment that "in case of unkown city it throws KeyError` (but honestly it would be even better to handle this behavior inside `get_weather` and for example return `None` in case city is not found. Because `KeyError` is about dictionaries, not about weather and user of the `get_weather` function should not be concerned about these implementation details.